hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/BackgroundContainerDataScanner.java

```diff
@@ -92,6 +92,13 @@ public void scanContainer(Container<?> c)
     if (result.isDeleted()) {
       LOG.debug("Container [{}] has been deleted during the data scan.", containerId);
     } else {
+      // Merkle tree write failure should not abort the scanning process. Continue marking the scan as completed.
+      try {
+        checksumManager.writeContainerDataTree(containerData, result.getDataTree());
+      } catch (IOException ex) {
+        LOG.error("Failed to write container merkle tree for container {}", containerId, ex);
```

Comment:
Same here, need to send an ICR if the checksum changes.

Reply:
This is handled in the latest commits.